pkg/build/metrics/prometheus/metrics.go

```diff
@@ -37,6 +39,11 @@ var (
 		[]string{"namespace", "name", "phase", "reason", "strategy"},
 		nil,
 	)
+	buildFinishedCounter = metrics.NewCounterVec(&metrics.CounterOpts{
+		Name:           buildFinishedQuery,
+		Help:           "Counts the total number of finished builds by namespace, buildConfig, phase, and build strategy",
+		StabilityLevel: metrics.ALPHA,
+	}, []string{"namespace", "buildconfig", "phase", "strategy"})
```

Comment:
@openshift/openshift-team-monitoring this adds a new (cluster-local) metric that has unbound cardinality. `namespace` and `buildconfig` are user-provided. `phase` and `strategy` have limited cardinality - 4 build strategies (`Source`, `Docker`, `Custom`, and `JenkinsPipeline`), 4 phases (`Finished`, `Failed`, `Error`, and `Cancelled`). Is this a problem?

If we were to export this metric to telemeter, I assume we'd need to create a recording rule to aggregate by phase and strategy. Is 16 too high?

Reply:
I'm not sure about the telemetry budget allocated for each team but 16 seems a bit high? cc @lilic 
As you pointed out, `buildconfig` and `namespace` labels seem contentious if you can't predict the maximum number of combinations.
One question to ask is how you're going to use this metric? Do these labels matter for alerting?

Reply:
First of all, thanks @adambkaplan for providing such details in the PR, its nice to see folks taking cardinality into account! :) 🎉 Regardless of telemtry we never want unbound cardinality in the first place for the sake of in cluster monitoring stack prometheus, as for telemetry the budget seems to be approx 10 series from my last information. Clayton should have more info.

Reply:
> One question to ask is how you're going to use this metric? Do these labels matter for alerting?

We plan to export the metric to Telemeter so that PM can gain insight into a) which build strategies are being used, and b) how successful end-developers are using builds. Namespace and name are not needed for alerting or anything like that.

Our primary challenge with builds is that they are like `CronJob/Job` objects - they have default history retention for successful and failed builds. Measuring just the state of the system will not give us an accurate measure of how well builds are working for end developers. Over time the ratio of successful:failed builds will reach 1:1 for a typical cluster. Hence, the counter that is incremented by the build controller.

We could change `phase` to be `result` and only have the labels `succeeded` and `failed`. This would bring us to within the cardinality budget while providing data that PM can digest at a high level. In that case I would change this metric name to `openshift_build_result_total`.

@lilic @simonpasquier I also did a review of our `openshift_build_*` metrics, and we have several overlapping metrics that are coming from openshift-state-metrics which likewise have unbound cardinality. It seems that adding `Counter` metrics there is not feasible due to the nature of its reconciliation and metrics collection.